main.py

```diff
@@ -78,7 +78,25 @@ def setCheckAttr(widget, var): self.settings.settings[var] = widget.isChecked()
 		def getCheckAttr(widget, var): widget.setChecked(self.settings.settings[var])
 		def setDropdownAttr(widget, var): self.settings.settings[var] = widget.currentText()
 		def getDropdownAttr(widget, var): widget.setCurrentText(self.settings.settings[var])
-		def updateLaunchCmd(): self.invocationPreview.setPlainText(Xqemu.launchCmdToString(Xqemu.generateLaunchCmd(self.settings, True)))
+
+		def updateLaunchCmd():
+			controller_arr = [
+				['controller_one', self.controller1],
+				['controller_two', self.controller2],
+				['controller_three', self.controller3],
+				['controller_four', self.controller4]
+			]
+			for i, controller in enumerate(controller_arr, 1):
+				for ab in ['A', 'B']:
```

Comment:
What does all of this have to do with `updateLaunchCmd`? This looks like a hack

Reply:
Sorry, I had forgotten about that note by the time I looked at this again.

I'd suggest to rename `updateLaunchCmd` to something more fitting then. Possibly splitting it into smaller routines - one for `updateLaunchCmd` and one for `hideUnavailableMemoryUnits`; or something along those lines.